sdk/python/feast/infra/offline_stores/remote.py

```diff
@@ -0,0 +1,173 @@
+import uuid
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Callable, List, Literal, Optional, Union
+
+import pandas as pd
+import pyarrow as pa
+import pyarrow.flight as fl
+import pyarrow.parquet
+from pydantic import StrictInt, StrictStr
+
+from feast import OnDemandFeatureView
+from feast.data_source import DataSource
+from feast.feature_logging import LoggingConfig, LoggingSource
+from feast.feature_view import FeatureView
+from feast.infra.offline_stores.offline_store import (
+    OfflineStore,
+    RetrievalJob,
+)
+from feast.infra.registry.base_registry import BaseRegistry
+from feast.repo_config import FeastConfigBaseModel, RepoConfig
+from feast.usage import log_exceptions_and_usage
+
+
+class RemoteOfflineStoreConfig(FeastConfigBaseModel):
+    type: Literal["remote"] = "remote"
+    host: StrictStr
+    """ str: remote offline store server port, e.g. the host URL for offline store  of arrow flight server. """
+
+    port: Optional[StrictInt] = None
+    """ str: remote offline store server port."""
+
+
+class RemoteRetrievalJob(RetrievalJob):
+    def __init__(
+        self,
+        config: RepoConfig,
+        feature_refs: List[str],
+        entity_df: Union[pd.DataFrame, str],
+        # TODO add missing parameters from the OfflineStore API
+    ):
+        # Generate unique command identifier
+        self.command = str(uuid.uuid4())
+        # Initialize the client connection
+        self.client = fl.connect(
+            f"grpc://{config.offline_store.host}:{config.offline_store.port}"
+        )
+        # Put API parameters
+        self._put_parameters(feature_refs, entity_df)
```

Comment:
I'm not 100% sure, but I think this should also go in `_to_arrow_internal` method. That's the behavior in other offline stores. Nothing is actually happening until the user calls one of the "action" methods on `RetrievalJob`.

Reply:
can you clarify more, what I understand you are asking to add  `self._put_parameters(feature_refs, entity_df)` into  ` function _to_arrow_internal`

Reply:
yes, take [bigquery](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/bigquery.py) engine for example. It's a bit more complicated, but note that `_upload_entity_df` (which essentially does the same thing as `self._put_parameters(feature_refs, entity_df)`) method is not called during the creation of RetrievalJob, it's actually part of `query_generator` function and is called from `_to_arrow_internal`/`_to_df_internal`.

Reply:
agree, we can move the parameter exchange a bit later in the flow. here we only wanted to create the flight at the request time, but there's actually no reason for it, it can surely be postponed